source/system-administrators/upgrade/upgrading-in-context-editing.rst

```diff
@@ -0,0 +1,221 @@
+
+.. code-block:: html
+
+    <#import "/templates/system/common/cstudio-support.ftl" as studio />
+
+Replace the ``cstudio-support.ftl`` import with the new ``crafter.ftl``,
+
+.. code-block:: html
+
+    <#import "/templates/system/common/crafter.ftl" as crafter />
+
+Remove the ``@studio.toolSupport`` macro call and add ``<@crafter.head/>`` to your ``head`` tag,
+``<@crafter.body_top/>`` to the top of your ``body`` tag and ``<@crafter.body_bottom/>`` to the
+bottom of your body tag.
+
+.. code-block:: html
+
+    <html>
+    <head>
+      <meta charset="utf-8">
+      <title>${model.title_t}</title>
+      <@crafter.head />
+    </head>
+    <body>
+      <@crafter.body_top/>
+      <main>
+        ...
+      </main>
+      <@crafter.body_bottom/>
+    </body>
+    </html>
+
+All of the invocations of the macros in ``cstudio-support.ftl`` need to be replaced. Examples of these are
+``@studio.iceAttr``, ``@studio.componentAttr``, ``@studio.componentContainerAttr``, etc. All of these need
+to be removed and replaced with the new system.
+
+The new system relies on you specifying each field of the model that's on the page on a per-element basis. Each
+tag in the page that prints out something from the CrafterCMS content type model should have its own element
+and that element needs to be marked with the relevant attributes which the new out of the box macros do for you.
+
+^^^^^^^^^^^^^
+Simple Values
+^^^^^^^^^^^^^
+
+Starting with simple values (e.g. strings printed on tags), locate each of the freemarker interpolation expressions
+(i.e. ``${contentModel.someFieldId}``). Make sure that the interpolation expression is by itself on a html tag.
+
+For example, ``By <span>${contentModel.authorName_s}</span>`` is correct since the interpolated value
+is alone in its container tag, as opposed to ``<span>By ${contentModel.authorName_s}</span>``, where
+there's a piece of text is inside the same element as the interpolation.
+
+Once you've located the interpolations and made sure each expression is alone in its element, replace
+the plain-html tag with a CrafterCMS macro.
+
+For example ``<span>${contentModel.authorName_s}</span>`` would become ``<@crafter.span>${contentModel.authorName_s}</@crafter.span>``.
+In addition to converting the plain tag to a macro, you most specify some additional metadata depending
+on the portion of the model that you're working with. Typically, interpolations refer to a field of
+the model, which is the missing piece of metadata on the latter example. Add the ``$field`` attribute
+to your model with the value being the field id of what you're printing.
+
+.. code-block:: html
+
+    <@crafter.span $field="authorName_s">${contentModel.authorName_s}</@crafter.span>
+
+Most HTML tags have an :ref:`equivalent macro <xbMacros>`, but if you happen to be using a *rare*
+or custom tag that doesn't have a macro, use you can use the ``@crafter.tag`` macro, which will print
```

Comment:
Typo?
   or custom tag that doesn't have a macro, ~use~ you can use the ``@crafter.tag`` macro,